ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MachineAccessIntegrationForm.tsx

```diff
@@ -0,0 +1,377 @@
+import * as yup from 'yup';
+import React, { ReactElement, useEffect, useState } from 'react';
+import {
+    Alert,
+    AlertVariant,
+    Button,
+    Flex,
+    FlexItem,
+    Form,
+    FormGroup,
+    FormSection,
+    PageSection,
+    Popover,
+    SelectOption,
+    TextInput,
+} from '@patternfly/react-core';
+import { FieldArray, FormikProvider } from 'formik';
+import { ArrowRightIcon, HelpIcon, PlusCircleIcon, TrashIcon } from '@patternfly/react-icons';
+import { IntegrationFormProps } from 'Containers/Integrations/IntegrationForm/integrationFormTypes';
+import useIntegrationForm from 'Containers/Integrations/IntegrationForm/useIntegrationForm';
+import FormMessage from 'Components/PatternFly/FormMessage';
+import FormLabelGroup from 'Containers/Integrations/IntegrationForm/FormLabelGroup';
+import IntegrationFormActions from 'Containers/Integrations/IntegrationForm/IntegrationFormActions';
+import FormSaveButton from 'Components/PatternFly/FormSaveButton';
+import FormCancelButton from 'Components/PatternFly/FormCancelButton';
+import SelectSingle from 'Components/SelectSingle';
+import { fetchRolesAsArray, Role } from 'services/RolesService';
+import { MachineConfigType } from 'services/MachineAccessService';
+
+export type MachineAccessConfig = {
+    id: string;
+    type: MachineConfigType;
+    tokenExpirationDuration: string;
+    mappings: {
+        key: string;
+        valueExpression: string;
+        role: string;
+    }[];
+    issuer: string;
+};
+
+export const validationSchema = yup.object().shape({
+    type: yup.string().trim().required('Type is required.'),
+    tokenExpirationDuration: yup.string().trim().required('Token expiration duration is required'),
+    mappings: yup.array().of(
+        yup.object().shape({
+            key: yup.string().trim().required('Key is required.'),
+            valueExpression: yup.string().trim().required('Value expression is required.'),
+            role: yup.string().trim().required('Role is required.'),
+        })
+    ),
```

Comment:
It appears that we need to ensure there is always at least one Rule line, so we should specify that in the validation schema. 

`.required()` used to enforce a non-empty array, but the current docs imply that no longer works; instead, something like `.min(1)` seems to be the new method.

```suggestion
    mappings: yup.array().of(
        yup.object().shape({
            key: yup.string().trim().required('Key is required.'),
            valueExpression: yup.string().trim().required('Value expression is required.'),
            role: yup.string().trim().required('Role is required.'),
        })
    ).min(1),
```